numerrin_wrapper/numerrin_wrapper.py

```diff
@@ -0,0 +1,189 @@
+        self.pool = NumerrinPool()
+        self.code = NumerrinCode(self.pool.ph)
+        self._meshes = {}
+        self.CM = DataContainer()
+        self.SP = DataContainer()
+        self.BC = DataContainer()
+        self.CM_extensions = {}
+        self._first = True
+
+    def run(self):
+        """Run Numerrin based on CM, BC and SP data
+
+        Raises
+        ------
+        Exception when solver not supported.
+
+        """
+
+        # put SP parameters to pool
+        for key in self.SP:
+            self.pool.put_variable(numname[key], self.SP[key])
+        # parse solver code
+        if self._first:
+            self.code.parse_string(
+                self.code.generate_code(self.CM,
+                                        self.SP,
+                                        self.BC,
+                                        self.CM_extensions))
+            self._first = False
+
+        # execute code
+        self.code.execute(1)
+        # save last iteration
+        for mesh in self.iter_meshes():
+            mesh._time = self.pool.get_variable('iteration')
+
+    def add_mesh(self, mesh):
+        """Add a mesh to the Numerrin modeling engine.
+
+        Parameters
+        ----------
+        mesh : ABCMesh
+            mesh to be added.
+
+        Returns
+        -------
+        proxy : NumerrinMesh
+            A proxy mesh to be used to update/query the internal representation
+            stored inside the modeling-engine. See get_mesh for more
+            information.
+
+        Raises
+        ------
+        Exception if mesh already exists
+
+        """
+
+        if mesh.name in self._meshes:
+            raise ValueError('Mesh \'{}\` already exists'.format(mesh.name))
+
+        self._meshes[mesh.name] = NumerrinMesh(mesh.name, mesh, self.pool)
+        return self._meshes[mesh.name]
+
+    def get_mesh(self, name):
+        """Get a mesh.
+
+        The returned mesh can be used to query and update the state of the
+        mesh inside the Numerrin modeling engine.
+
+        Parameters
+        ----------
+        name : str
+            name of the mesh to be retrieved.
+
+        Returns
+        -------
+        NumerrinMesh
+
+        """
+
+        return self._meshes[name]
+
+    def iter_meshes(self, names=None):
+        """Returns an iterator over a subset or all of the meshes.
+
+        Parameters
+        ----------
+        names : list of str
+            names of specific meshes to be iterated over.
+            If names is not given, then all meshes will
+            be iterated over.
+
+        Returns
+        ----------
+        Iterator over a subset or all of the meshes
+
+        Raises
+        ------
+        Exception if some mesh fron mesh names list not found
+
+        """
+
+        if names is None:
+            for name in self._meshes:
+                yield self._meshes[name]
+        else:
+            for name in names:
+                if name in self._meshes:
```

Comment:
``if-else`` statement not necessary, i.e. ``yield self._meshes[name]`` is enough
(the mesh dictionary will raise an exception, if the ``name`` is invalid).